corehq/apps/es/es_query.py

```diff
@@ -466,6 +466,12 @@ def values_list(self, *fields, **kwargs):
     def count(self):
         return self.adapter.count(self.raw_query)
 
+    def exists(self):
+        """Checks to see whether any documents match the query"""
+        query = self.size(0)
+        query.es_query['terminate_after'] = 1
```

Comment:
I don't think this pattern/idiom of setting terminate_after = 1 is widely used in the codebase (at least I'm not familiar with it. Maybe a short in-line comment explaining it would be useful? My read is that it's a kind of heuristic optimization that could technically result in returning `False` (doesn't exist) when the right answer is `True` (does exist). Is that true, and is that okay?